app/main.py

```diff
@@ -83,6 +88,95 @@ def _get_client(self):
+            self._api_instance = client.CoreV1Api()
+
+        return self._api_instance
+
+    def _get_kubernetes_secrets_list(self):
+        if self._secrets_list is None:
+            api_instance = self._get_kubernetes_api_instance()
+            api_response = api_instance.list_namespaced_secret(namespace=self._secrets_namespace)
+            
+            secret_name_list = []
+            should_continue = True
+
+            while should_continue is True:
+                continue_value = api_response.metadata._continue
+                secrets_list = api_response.items
+                for item in secrets_list:
+                    secret_name_list.append(item.metadata.name)
+
+                if continue_value is not None:
+                    api_response = api_instance.list_namespaced_secret(namespace=self._secrets_namespace, _continue = continue_value)
+                else:
+                    should_continue = False
+
+            self._secrets_list = secret_name_list
+        
+        return self._secrets_list
+
+    def _create_kubernetes_secret_objects(self, key, value):
+        key = key.lower()
+        api_instance = self._get_kubernetes_api_instance()
+        secret = client.V1Secret()
+        encoded_secret = base64.b64encode(bytes(value))
+
+        secret.metadata = client.V1ObjectMeta(name=key)
+        secret.type = "Opaque"
+        secret.data = { "secret" : encoded_secret }
+
+        secrets_list = self._get_kubernetes_secrets_list()
+
+        _logger.info('Creating or updating Kubernetes Secret object: %s', key)
+        try:
+            if key in secrets_list:
+                api_instance.patch_namespaced_secret(name=key, namespace=self._secrets_namespace, body=secret)
+            else:
+                api_instance.create_namespaced_secret(namespace=self._secrets_namespace, body=secret)
+        except:
+            _logger.exception("Failed to create or update Kubernetes Secret")
+
+    def grab_secrets_kubernetes_objects(self):
+        """
+        Gets secrets from KeyVault and creates them as Kubernetes secrets objects
+        """
+        vault_base_url = os.getenv('VAULT_BASE_URL')
+        secrets_keys = os.getenv('SECRETS_KEYS')
+        self._secrets_namespace = os.getenv('SECRETS_NAMESPACE')
+
+        if self._secrets_namespace is None:
+            self._secrets_namespace = "default"
+
+        client = self._get_client()
+        _logger.info('Using vault: %s', vault_base_url)
+
+        # Retrieving all secrets from Key Vault if specified by user
+        if secrets_keys is None:
+            _logger.info('Retrieving all secrets from Key Vault.')
+
+            secrets_keys = ""
+            all_secrets = list(client.get_secrets(vault_base_url))
+            key_list = []
+            for secret in all_secrets:
+                split = secret.id.split('/')
+                key = split[len(split) - 1]
+                key_list.append(key)
```

Comment:
where are you using `key_list`?

in addition, you could simplify this code with a one-liner - 
```python
secret_keys = ';'.join([secret.id.split('/')[-1] for secret in all_secrets])
```

Reply:
Looks like I wasn't using it, switched to your one-liner